lib/msal-browser/test/cache/BrowserCacheManager.spec.ts

```diff
@@ -325,7 +326,532 @@ describe("BrowserCacheManager tests", () => {
+            await browserLocalStorage.clear();
+        });
+
+        it("setTemporaryCache", () => {
+            browserSessionStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserLocalStorage.setTemporaryCache("cacheKey2", cacheVal, true);
+            expect(window.sessionStorage.getItem(msalCacheKey)).toBe(cacheVal);
+            expect(window.localStorage.getItem(msalCacheKey2)).toBe(cacheVal);
+        });
+
+        it(
+            "getTemporaryCache falls back to local storage if not found in session/memory storage",
+            () => {
+                const testTempItemKey = "test-temp-item-key";
+                const testTempItemValue = "test-temp-item-value";
+                window.localStorage.setItem(testTempItemKey, testTempItemValue);
+                cacheConfig.cacheLocation = BrowserCacheLocation.LocalStorage;
+                browserLocalStorage = new BrowserCacheManager(TEST_CONFIG.MSAL_CLIENT_ID, cacheConfig, browserCrypto, logger);
+                expect(browserLocalStorage.getTemporaryCache(testTempItemKey)).toBe(testTempItemValue);
+            }
+        )
+
+        it("setItem", () => {
+            window.sessionStorage.setItem(msalCacheKey, cacheVal);
+            window.localStorage.setItem(msalCacheKey2, cacheVal);
+            expect(browserSessionStorage.getItem(msalCacheKey)).toBe(cacheVal);
+            expect(browserLocalStorage.getItem(msalCacheKey2)).toBe(cacheVal);
+        });
+
+        it("removeItem()", () => {
+            browserSessionStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserLocalStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserSessionStorage.removeItem(msalCacheKey);
+            browserLocalStorage.removeItem(msalCacheKey);
+            expect(window.sessionStorage.getItem(msalCacheKey)).toBeNull();
+            expect(window.localStorage.getItem(msalCacheKey)).toBeNull();
+            expect(browserLocalStorage.getTemporaryCache("cacheKey", true)).toBeNull();
+            expect(browserSessionStorage.getTemporaryCache("cacheKey", true)).toBeNull();
+        });
+
+        it("containsKey()", () => {
+            browserSessionStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserLocalStorage.setItem(msalCacheKey, cacheVal);
+            expect(browserSessionStorage.containsKey(msalCacheKey)).toBe(true);
+            expect(browserLocalStorage.containsKey(msalCacheKey)).toBe(true);
+        });
+
+        it("clear()", async () => {
+            browserSessionStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            browserLocalStorage.setTemporaryCache("cacheKey", cacheVal, true);
+            await browserSessionStorage.clear();
+            await browserLocalStorage.clear();
+            expect(browserSessionStorage.getKeys()).toHaveLength(0);
+            expect(browserLocalStorage.getKeys()).toHaveLength(0);
+        });
+
+        describe("Getters and Setters", () => {
```

Comment:
Tests below this line don't appear relevant to the changes in this PR. Am I missing something?

Reply:
I have copy-pasted these tests from the existing "Interface functions" tests. But in this test setup I changed the cacheConfig for the temporaryCacheLocation to LocalStorage. The reasoning behind this was to show that the existing functionality doesn't break with the temporaryCacheLocation to LocalStorage. If you prefer having them removed. Let me know